src/index/client/openid/identity.ts

```diff
@@ -61,6 +62,56 @@ export function decodeIdToken(accessTokenResponse: OpenIDAccessTokenResponse) {
 
 }
 
+/**
+ * Verify the `id_token` from an access token response.
+ *
+ * @param accessTokenResponse The access token response.
+ * @param [jwkPemFormatMap] Map of the retrieved JSON Web Keys, keyed by the Key ID Parameter, in PEM format.
+ */
+export function verifyIdToken(accessTokenResponse: OpenIDAccessTokenResponse, jwkPemFormatMap?: JsonWebKeyPemFormatMap) {
+
+	if (!jwkPemFormatMap) {
+		throw new Error('Unable to verify ID token: No JWKs provided');
+	}
+
+	if (accessTokenResponse.id_token) {
+
+		if (typeof accessTokenResponse.id_token !== 'string') {
+			throw new Error('Unable to verify ID token: id_token is not a string');
+		}
+
+		const decodedIdToken = decode(accessTokenResponse.id_token, {
+			complete: true
+		});
+
+		if (!decodedIdToken || typeof decodedIdToken !== 'object') {
+			throw new Error('Unable to verify ID token: decoded token is not an object');
+		}
+
+		if (!decodedIdToken.header) {
+			throw new Error('Unable to verify ID token: decoded token does not contain the header');
+		}
+
+		if (!decodedIdToken.header.kid) {
+			throw new Error('Unable to verify ID token: decoded token header does not contain the kid');
+		}
+
+		const secret = jwkPemFormatMap[decodedIdToken.header.kid];
```

Comment:
It's not very secret :-), maybe say public key?

Reply:
Quite right 👍 Fixed.